frontend/src/screens/ProductScreen.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import Rating from '../components/Rating';
+import data from '../data';
+
+export default function ProductScreen(props) {
+  const product = data.products.find((x) => x._id === props.match.params.id);
```

Comment:
If anybody have a problem of "Product Not Found" please check this line:
const product = data.products.find((x) => Number(x._id) === Number(props.match.params.id));

Reply:
i got this error:   Uncaught TypeError: Cannot read properties of undefined (reading 'params').

Reply:
This code works )) 
 const { id } = useParams();
  const product = data.products.find((x) => x._id === Number(id));
Don't forget to
 import { Link, useParams } from "react-router-dom";

Reply:
> This code works )) const { id } = useParams(); const product = data.products.find((x) => x._id === Number(id)); Don't forget to import { Link, useParams } from "react-router-dom";

still showing  product not fount 

find code below

import React from 'react';
import { Link, useParams } from "react-router-dom";
// import Rating from '../components/Rating';
import data from '../data';

export default function ProductScreen(props) {
    const { id } = useParams();
    const product = data.products.find((x) => x._id === Number(id));
  if (!product) {
    return <div> Product Not Found</div>;
  }
  return (
    <div>
        Product Found !
    </div>
  )
}